### PR TITLE
df/add-stream-test-vectors

### PR DESCRIPTION
**Pull request by sappenin**
Introduces a JSON file that allows STREAM implementations to validate their ASN.1 OER encoding/decoding of alll Frames defined in IL-RFC-29.

Signed-off-by: sappenin <sappenin@gmail.com>

### CODE REVIEW

File: 0029-stream/test-vectors/README.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Stream Frame Test Vectors
+In order to verify different implementations of the STREAM specification, we provide a series of test vectors in JSON format to compare expected vs actual ASN.1 OER encoding/decoding values.
+
+## Valid Test Vectors
+The file [ValidStreamFrameTestVectors.json](https://github.com/interledger/rfcs/tree/master/0029-stream/test-vectors/ValidStreamFrameTestVectors.json) contains a JSON array of elements representing Stream Frame values in addition to a corresponding Base64-encoded binary reprsentation of the ASN.1 OER bytes for a given frame.
```
Comment (nhartner) [inline 0029-stream/test-vectors/README.md]: url 404s. Or is that something that is still to be created?

Comment (nhartner) [reply 0029-stream/test-vectors/README.md]: Never mind needed to look farther down in the PR

**Comment by sentientwaffle** [comment]
@sappenin consider using part or all of https://github.com/interledgerjs/ilp-protocol-stream/blob/master/test/fixtures/packets.json (generated by https://github.com/interledgerjs/ilp-protocol-stream/blob/master/scripts/generate-fixtures.ts). It might be excessive, but it does cover some important gotchas around e.g. exceeding max uint64s (https://github.com/interledgerjs/ilp-protocol-stream/blob/master/test/fixtures/packets.json#L810-L846).

**Comment by sappenin** [comment]
> @sappenin consider using part or all of https://github.com/interledgerjs/ilp-protocol-stream/blob/master/test/fixtures/packets.json 

@sentientwaffle Wow, that test harness is _much_ better than the one I created. I see there's a generator, but is it correct to say that we can just take the `packets.json` and use that as a fixed fixture, or is there some important reason to re-generate fixtures on every test run?

**Comment by sentientwaffle** [comment]
@sappenin the generator is just for convenience -- to make it easier to add additional test cases. Adding the `packets.json` file is sufficient, there's no need to run (or even include) the generator script.

**Comment by sappenin** [comment]
@sentientwaffle Sounds good - one question about the `frame:connection_new_address:empty` test-case. Is that something that should be tolerated? This is the only thing that the Java code barfs on, but I'm thinking that a `ConnectionNewAddress` frame with not address should be a protocol violation, no?

**Comment by sentientwaffle** [comment]
You're correct, an empty address should be invalid. Not sure what I was thinking when I added that test case.

**Comment by sappenin** [comment]
@sentientwaffle I just pushed new changes to this PR that mirror your test vectors. Want to give this another look?

File: 0029-stream/test-vectors/README.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
+# Stream Frame Test Vectors
+In order to verify different implementations of the STREAM specification, we provide a series of test vectors in JSON format to compare expected vs actual ASN.1 OER encoding/decoding values.
+
+## Valid Test Vectors
+The file [StreamPacketFixtures.json](https://github.com/interledger/rfcs/tree/master/0029-stream/test-vectors/StreamPacketFixtures.json) contains a JSON array of elements representing Stream packets with embedded Stream Frame values. The JSON file also includes a corresponding Base64-encoded binary representation of the ASN.1 OER bytes for a given packet/frame combination.
+ 
+ Tests should be written to validate that encoded Stream frames match the expected binary output in the JSON file. Conversely, tests should also be written to validate that decoded binary values match the values in the JSON test vector file.
```
Comment (sentientwaffle) [inline 0029-stream/test-vectors/README.md]: nit: `s/frames/packets`

File: 0029-stream/test-vectors/README.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+   
+ Each array element in the JSON file contains an object whose values have the the following meanings:
```
Comment (sentientwaffle) [inline 0029-stream/test-vectors/README.md]: nit: 'properties' or 'fields' is a little less ambiguous than 'values'.

File: 0029-stream/test-vectors/README.md [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+| Field | Description |
+|----:|:----|
+| `name` | The name of the test vector. |
+| `packet` | A JSON object containing Stream Packet information. |
+| `buffer` | A string containing the Base64-encoded bytes that are expected when the Stream packet and frame are encoded according to the rules of [IL-RFC-29](https://github.com/interledger/rfcs/blob/master/0029-stream/0029-stream.md). |
```
Comment (sentientwaffle) [inline 0029-stream/test-vectors/README.md]: typo: `s/frame/frames`

File: 0029-stream/test-vectors/README.md (hunk 2 above)
Comment (sappenin) [reply 0029-stream/test-vectors/README.md]: Fixed via 7abe5e63b7190471af08bf6a71fc3feff6ab2daf

File: 0029-stream/test-vectors/README.md [hunk 5]
```diff
@@ -2,19 +2,29 @@
...
-* `expectedAsn1OerBytes`: A string containing the Base64-encoded bytes that are expected when the specified Stream Frame is encoded into its ASN.1 OER encoding.
-* `Misc Fields`: Each Stream Frame has its own particular fields. For example, the `ConnectionNewAddress` frame has a `sourceAddress` property, whereas other frames do not. The JSON file contains a sample value of each property available in a particular frame.
+| Field | Description |
+|----:|:----|
+| `name` | The name of the test vector. |
+| `packet` | A JSON object containing Stream Packet information. |
+| `buffer` | A string containing the Base64-encoded bytes that are expected when the Stream packet and frame are encoded according to the rules of [IL-RFC-29](https://github.com/interledger/rfcs/blob/master/0029-stream/0029-stream.md). |
+| `packet.sequence` | The Stream packet's sequence number |
+| `packet.packetType` | The type of Stream packet represented in the test vector. |
+| `packet.packetType` | The type of Stream packet represented in the test vector. |
```
[the remark this answers is not in the document]
Comment (sappenin) [reply 0029-stream/test-vectors/README.md]: Fixed via 7abe5e63b7190471af08bf6a71fc3feff6ab2daf

File: 0029-stream/test-vectors/README.md [hunk 6]
```diff
@@ -2,19 +2,29 @@
...
 
 ## Test Validations
 Each implementation of STREAM should work across the test vectors defined above. The following is a link to the test coverage in various projects (please submit a PR to this file if you have an implementation not defined in this list):
 
-* Java Stream: [ValidStreamFrameVectorsTest](https://github.com/hyperledger/quilt/blob/master/codecs-parent/codecs-stream/src/test/java/org/interledger/codecs/stream/frame/ValidStreamFrameVectorsTest.java)
-* Rust Stream: TBD.
-* Javascript Stream: TBD.
+* Java (Hyperledger Quilt): [Fixtures](https://github.com/hyperledger/quilt/tree/master/codecs-parent/codecs-stream/src/test/resources/StreamPacketFixtures.json) | [Test Class](https://github.com/hyperledger/quilt/tree/master/codecs-parent/codecs-stream/src/test/java/org/interledger/codecs/stream/StreamPacketFixturesTest.java)
+* Rust Stream: `TBD`.
+* Javascript Stream: `TBD`.
```
[the remark this answers is not in the document]
Comment (sappenin) [reply 0029-stream/test-vectors/README.md]: Fixed via 7abe5e63b7190471af08bf6a71fc3feff6ab2daf

**Pull request merged**